src/30_Advanced/SeatMap.html

```diff
@@ -0,0 +1,202 @@
+          [[range .Seats]] 
+           <rect class="seat unavailable" 
+                           x="[[.X]]" y="[[.Y]]" width="[[.Width]]" height="[[.Height]]"
+                          role="button"
+                          tabindex="0" rx="[[.Rx]]" ry="[[.Ry]]"/>   
+            [[end]]
+              
+        </svg>
+      </div>
+          <template type="amp-mustache">
+          <amp-pan-zoom layout="fill" id="pan-zoom"
+            on="transformEnd:AMP.setState({seatmap: {scale: event.scale, x: event.x, y: event.y, zoom: event.scale <= 1 ? 0 : 1}})">
+            <svg class="seat-map-svg" preserveAspectRatio="xMidYMin slice" viewBox="0 0 {{width}} {{height}}">
+                {{#seats}}                
+                {{#standard }}
+                    <!--~
+                    If a seat is selected, when the user selects it again, it's removed from the seatmap.selectedSeats list
+                    ~-->
+                    <rect class="seat selected"
+                          hidden [hidden]="!seatmap.selectedSeats.includes('{{id}}')"
+                          x="{{x}}" y="{{y}}" width="{{width}}" height="{{height}}"
+                          on="tap:AMP.setState({
+                            seatmap: {selectedSeats: copyAndSplice(seatmap.selectedSeats, seatmap.selectedSeats.indexOf('{{id}}'), 1)}
+                          })"
+                          role="button"
+                          tabindex="0" rx="{{rx}}" ry="{{ry}}"></rect>
+                    {{#available }}
+                      <rect class="seat" 
+                          x="{{x}}" y="{{y}}" width="{{width}}" height="{{height}}"
+                          [hidden]="seatmap.selectedSeats.includes('{{id}}')"
+                          on="tap:AMP.setState({
+                            lastSelectedSeat: '{{id}}',
+                            seatmap: {selectedSeats: seatmap.selectedSeats.concat('{{id}}')}
+                          })"
+                          role="button"
+                          tabindex="0" rx="{{rx}}" ry="{{ry}}"></rect>  
+                    {{/available }}
+                    {{^available}}
+                          <rect class="seat unavailable" 
+                          x="{{x}}" y="{{y}}" width="{{width}}" height="{{height}}"
+                          role="button"
+                          tabindex="0" rx="{{rx}}" ry="{{ry}}">   
+                          </rect>            
+                    {{/available }}
+                  {{/standard }}
+                {{/seats}}
+              </svg>
+            </amp-pan-zoom>
+          </template>
+        </amp-list>
+      </div>
+      <!-- To enable desktop users, we provide + and - buttons. We use `amp-bind` to decide whether to show the + or - button depending
+      on the state of the zooming -->
+      <div>
+        <button class="zoom-button" on="tap:pan-zoom.transform(scale=1)" hidden [hidden]="seatmap.zoom==0">-</button>
```

Comment:
This is no longer needed is it?

Reply:
at the moment yes because latest changes are not in prod.